### PR TITLE
[quant][refactor tests] Split test_quantize into test_quantize_eager_ptq, test_quantize_eager_qat and test_fusion

### PR DESCRIPTION
**Pull request by supriyar**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #58963 [quant][refactor tests] split test_workflow_module into test_workflow_ops and test_workflow_module
* #58943 [quant][refactor tests] Move test_quantized_module into test_quantize_eager_ptq
* #58929 [quant][refactor tests] Move test_quantized_functional.py into test_quantize_eager_ptq.py
* #58928 [quant][refactor tests] Move test_qat_module into test_quantize_eager_qat
* **#58927 [quant][refactor tests] Split test_quantize into test_quantize_eager_ptq, test_quantize_eager_qat and test_fusion**

Summary:
Part of larger re-factor of quantization tests to make it clearer as to which test belongs where.

proposed folder structure
```
test/quantization
         - bc/
            - test_backward_compatibility.py
         - core/
            - test_quantized_kernels.py
            - test_quantized_workflow_ops.py
            - test_quantized_tensor.py
            - test_workflow_module.py
         - eager/
            - test_quantize_eager_ptq.py
            - test_quantize_eager_qat.py
            - test_fusion.py
         - equalization/
            - test_equalize_eager.py
            - test_bias_correction_eager.py
         - fx/
           - test_quantize_fx.py
         - jit/
            - test_quantize_jit.py
            - test_fusion_passes.py
         - numeric_suite/
            - test_numeric_suite_fx.py
            - test_numeric_suite_eager.py
```
Test Plan:
python test/test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28683926](https://our.internmc.facebook.com/intern/diff/D28683926)

### CODE REVIEW

**Comment by supriyar** [comment]
@supriyar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28683926).

File: test/quantization/test_quantize_eager_ptq.py [hunk 1]
```diff
@@ -1989,343 +1349,6 @@ def test_lower_graph_conv3d(self):
...
-                return torch.ops.quantized.linear_unpack(self._packed_weight)[0]
-
-            @weight.setter
-            def weight(self, w):
-                self._packed_weight = torch.ops.quantized.linear_prepack(w)
-
-        with torch._jit_internal._disable_emit_hooks():
-            x = torch.jit.script(Linear(10, 10))
-            torch._C._jit_pass_erase_shape_information(x.graph)
-
```
Comment (raghuramank100) [inline test/quantization/test_quantize_eager_ptq.py:2328]: Are these tests deleted?

File: test/test_determination.py [hunk 2]
```diff
@@ -74,7 +74,7 @@ def test_test_file(self):
             ["distributed/rpc/test_process_group_agent"],
         )
         self.assertEqual(
-            self.determined_tests(["test/quantization/test_quantize.py"]),
+            self.determined_tests(["test/quantization/test_quantize_eager_ptq.py"]),
```
Comment (raghuramank100) [inline test/test_determination.py:77]: Why is this specific file chosen instead of test_quantize?

**Review by raghuramank100 — APPROVED** [review]
Looks good, but had a few comments

File: test/test_determination.py (hunk 2 above)
Comment (supriyar) [reply test/test_determination.py:77]: We deleted test_quantize in the PR, so I used another test for this. I think it doesn't matter which specific one we use here.

File: test/quantization/test_quantize_eager_ptq.py (hunk 1 above)
Comment (supriyar) [reply test/quantization/test_quantize_eager_ptq.py:2328]: Yes, these were under TestDeprecatedJitQuantized so I thought we don't support them anymore. I can add them back.

**Comment by supriyar** [comment]
@supriyar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28683926).

**Pull request closed without merging**